### PR TITLE
[benchmark] Initial commit for Google Benchmark library

### PR DESCRIPTION
**Pull request by KindDragon**

### CODE REVIEW

**Comment by ras0219-msft** [comment]
Looks great, thanks!

**Comment by ras0219-msft** [comment]
Actually wait, I just realized this adds a `benchmark::` cmake export.

The full explanation is inside #616, (specifically https://github.com/Microsoft/vcpkg/issues/616#issuecomment-276853511) but basically we can't add cmake exports that could conflict with future choices by the upstream library owner. Exports like this are _essentially_ adding a new "feature/api" (it just happens to be at the buildsystem level instead of the source code level), which we strongly want to avoid because of potential future breakages.

As part of the #616 discussion, we _should_ consider automatically creating CMake config files which will
- Work uniformly and correctly across *all* Vcpkg packages. We can do this because we have strict layout requirements.
- Be cheap to maintain and not require patching every library
- Have no chance of conflicting with any export choices that upstream libraries decide to make -- if Google Benchmark decides to start exporting the `Benchmark::libbenchmark` target tomorrow, we will be able to simply add upstream's support and not worry about conflicting with our automatically generated files.

As a strawman example, we might add `find_package(vcpkg-X)` for all ports `X`. However, this should be further discussed in the other thread.

tldr; we can't add the export. I will remove it during the merge when I get time.

**Comment by KindDragon** [comment]
> Actually wait, I just realized this adds a benchmark:: cmake export.

They already add this name in `master` https://github.com/google/benchmark/blob/master/src/CMakeLists.txt#L45

**Comment by ras0219-msft** [comment]
**awesome**!

Thanks for the clarification!

**Comment by verbit** [comment]
The google-benchmark master did add the target export `benchmarkTargets` as well as the accompanying `benchmarkConfig` and `benchmarkVersion`. However, this PR seems only to add `benchmarkTargets`.

 Wouldn't it make sense to include the other two files, too? In addition, this enables importing the library via `find_package`.
